src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Compilation.cs

```diff
@@ -133,7 +133,9 @@ public virtual ISymbolNode GetFieldRvaData(FieldDesc field)
             }
             else if (field is ExternSymbolMappedField externField)
             {
-                return NodeFactory.ExternSymbol(externField.SymbolName);
+                string externName = externField.SymbolName;
+                externName = NodeFactory.NameMangler.NodeMangler.ExternVariable(externName);
+                return NodeFactory.ExternSymbol(externName);
```

Comment:
It seems like it would make for cleaner code to have some `NodeFactory.ExternVariable` that mangles appropriately, so that we could write just:
```
return NodeFactory.ExternVariable(externField.SymbolName);
```
(Applies to a couple other places too).

Being able to differentiate between code and data symbols would be useful downstream too.